src/data/transforms/minmax.rs

```diff
@@ -0,0 +1,247 @@
+//!
+//! // Constructs a new `MinMaxScaler` to map minimum to 0 and maximum
+//! // to 1.
+//! let mut transformer = MinMaxScaler::default();
+//!
+//! let inputs = Matrix::new(2, 2, vec![-1.0, 2.0, 1.5, 3.0]);
+//!
+//! // Transform the inputs to get output data with required minimum
+//! // and maximum.
+//! let transformed = transformer.transform(inputs).unwrap();
+//! ```
+
+use learning::error::{Error, ErrorKind};
+use linalg::Matrix;
+use super::Transformer;
+
+use libnum::Float;
+
+/// The `MinMaxScaler`
+///
+/// The `MinMaxScaler` provides an implementation of `Transformer`
+/// which allows us to transform the input data to have a new minimum
+/// and maximum per column.
+///
+/// See the module description for more information.
+#[derive(Debug)]
+pub struct MinMaxScaler<T: Float> {
+    /// Values to scale each column by
+    scale_factors: Option<Vec<T>>,
+    /// Values to add to each column after scaling
+    const_factors: Option<Vec<T>>,
+    /// The min of the new data (default 0)
+    scaled_min: T,
+    /// The max of the new data (default 1)
+    scaled_max: T,
+}
+
+/// Create a default `MinMaxScaler` with minimum of `0` and
+/// maximum of `1`.
+impl<T: Float> Default for MinMaxScaler<T> {
+    fn default() -> MinMaxScaler<T> {
+        MinMaxScaler {
+            scale_factors: None,
+            const_factors: None,
+            scaled_min: T::zero(),
+            scaled_max: T::one(),
+        }
+    }
+}
+
+impl<T: Float> MinMaxScaler<T> {
+    /// Constructs a new MinMaxScaler with the specified scale.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rusty_machine::data::transforms::{MinMaxScaler, Transformer};
+    ///
+    /// // Constructs a new `MinMaxScaler` which will give the data
+    /// // minimum `0` and maximum `2`.
+    /// let transformer = MinMaxScaler::new(0.0, 2.0);
+    /// ```
+    pub fn new(min: T, max: T) -> MinMaxScaler<T> {
+        MinMaxScaler {
+            scale_factors: None,
+            const_factors: None,
+            scaled_min: min,
+            scaled_max: max,
+        }
+    }
+}
+
+impl<T: Float> Transformer<Matrix<T>> for MinMaxScaler<T> {
+    fn transform(&mut self, mut inputs: Matrix<T>) -> Result<Matrix<T>, Error> {
+        let features = inputs.cols();
+
+        let mut input_min = vec![T::max_value(); features];
+        let mut input_max = vec![T::min_value(); features];
+
+        for row in inputs.iter_rows() {
+            for (idx, feature) in row.into_iter().enumerate() {
+                if !feature.is_finite() {
+                    return Err(Error::new(ErrorKind::InvalidData,
+                                          format!("Data point in column {} cannot be \
+                                                   processed",
+                                                  idx)));
+                }
+
+                if *feature < input_min[idx] {
+                    input_min[idx] = *feature;
+                }
```

Comment:
Very small point: 
You can avoid bound checks and `enumerate` using `zip` ... and only one zip if you work with tuples.
```rust
let mut input_min_max = vec![(T::min_value(), T::masx_value()); features];
```
... and computing the `scales` is simpler as well:
```rust
let scales = try!(input_min_max.iter().map(|(x, y)| {
    let s = (self.scaled_max - self.scaled_min) / (y - x);
    if s.is_finite() { 
        Ok(s) 
    } else {  
        Err(Error::new(ErrorKind::InvalidData,"Constant feature columns not supported."))
    }
}).collect::<Result<Vec<_>, _>>());
```